### PR TITLE
Support for multimedia (#266)

### PR DESCRIPTION
**Pull request by ben-pr-p**
Hey!

This adds support for images in texts. The syntax is, in any message, `[https://image-url.com/image-is-here]`.

screen shot 2017-12-03 at 2 19 54 pm

And the result:

img_0122

### CODE REVIEW

File: src/server/api/lib/twilio.js [hunk 1]
```diff
@@ -109,15 +109,26 @@ async function sendMessage(message) {
     if (message.service !== 'twilio') {
       log.warn('Message not marked as a twilio message', message.id)
     }
-    const messageParams = {
+
+    // Image extraction
+    const extractor = new RegExp(/\[.*http(.*)\]/)
```
Comment (schuyler1d) [inline src/server/api/lib/twilio.js]: I think the first `.*` should be `\s*` and the second you probably want `([^\]]*)` to be a little safer

**Review by schuyler1d — CHANGES_REQUESTED** [review]
:+1: 
after we tweak the regex.  This is great!

**Review by schuyler1d — APPROVED** [review]

File: src/server/api/lib/twilio.js [hunk 2]
```diff
@@ -109,15 +109,26 @@ async function sendMessage(message) {
     if (message.service !== 'twilio') {
       log.warn('Message not marked as a twilio message', message.id)
     }
-    const messageParams = {
+
+    // Image extraction
+    const extractor = new RegExp(/\[\s*(http[^\]\s]*)\s*\]/)
```
Comment (jmcarp) [inline src/server/api/lib/twilio.js:114]: WDYT about moving this to a separate function and adding tests?

**Comment by shakalee14** [comment]
this is ready to be merged and was qa'd on stage

**Comment by joemcl** [comment]
I'm trying to send a small test campaign with an image - not working so far. 
The number I'm sending is MMS-enabled in Twiio, just checked. 
The Twilio message log  shows an unhelpful error message - (Error:30008)Unknown error. 
The link I'm using is [https://drive.google.com/file/d/1i6X8OmTqCLEdbhTxXHbnaZFvmTZ7StNQ/view?usp=sharing], that's OK, just tried that in an 'incognito window' . 
Checking if there's some Heroku setting I need to change/tweak. 
I'm using our staging Spoke instance, it works fine for test campaigns sending plain text.

**Comment by joemcl** [comment]
Thanks @bchrobot *shouldn't* matter - the link and image is public - but of course it might :) cause  Google. So I'll check out imgur.com. In the test above, is that i.ytimg.com link produced by imgur.com or, like, YouTube? Some other site?

**Comment by joemcl** [comment]
OK @bchrobot thanks, so this link worked, hosted on imgur - https://i.imgur.com/lv66Kzg.jpg - a 'Direct Link'. 
An imgur 'Image Link' like https://imgur.com/lv66Kzg (does not contain file type suffix .jpg) did not. 
Thanks!

**Pull request merged**